Build fails with icu77 due to icu-uc not being in the include path

Linking fails with
```
c++  -o src/libkiwix.so.13.1.0 src/libkiwix.so.13.1.0.p/meson-generated_.._.._static_libkiwix-resources.cpp.o src/libkiwix.so.13.1.0.p/meson-generated_.._.._static_libkiwix-i18n-resources.cpp.o src/libkiwix.so.13.1.0.p/book.cpp.o src/libkiwix.so.13.1.0.p/bookmark.cpp.o src/libkiwix.so.13.1.0.p/library.cpp.o src/libkiwix.so.13.1.0.p/manager.cpp.o src/libkiwix.so.13.1.0.p/libxml_dumper.cpp.o src/libkiwix.so.13.1.0.p/opds_dumper.cpp.o src/libkiwix.so.13.1.0.p/html_dumper.cpp.o src/libkiwix.so.13.1.0.p/library_dumper.cpp.o src/libkiwix.so.13.1.0.p/downloader.cpp.o src/libkiwix.so.13.1.0.p/server.cpp.o src/libkiwix.so.13.1.0.p/search_renderer.cpp.o src/libkiwix.so.13.1.0.p/subprocess.cpp.o src/libkiwix.so.13.1.0.p/aria2.cpp.o src/libkiwix.so.13.1.0.p/tools_base64.cpp.o src/libkiwix.so.13.1.0.p/tools_pathTools.cpp.o src/libkiwix.so.13.1.0.p/tools_regexTools.cpp.o src/libkiwix.so.13.1.0.p/tools_stringTools.cpp.o src/libkiwix.so.13.1.0.p/tools_networkTools.cpp.o src/libkiwix.so.13.1.0.p/tools_opdsParsingTools.cpp.o src/libkiwix.so.13.1.0.p/tools_languageTools.cpp.o src/libkiwix.so.13.1.0.p/tools_otherTools.cpp.o src/libkiwix.so.13.1.0.p/tools_archiveTools.cpp.o src/libkiwix.so.13.1.0.p/kiwixserve.cpp.o src/libkiwix.so.13.1.0.p/name_mapper.cpp.o src/libkiwix.so.13.1.0.p/server_byte_range.cpp.o src/libkiwix.so.13.1.0.p/server_etag.cpp.o src/libkiwix.so.13.1.0.p/server_request_context.cpp.o src/libkiwix.so.13.1.0.p/server_response.cpp.o src/libkiwix.so.13.1.0.p/server_internalServer.cpp.o src/libkiwix.so.13.1.0.p/server_internalServer_catalog.cpp.o src/libkiwix.so.13.1.0.p/server_i18n.cpp.o src/libkiwix.so.13.1.0.p/opds_catalog.cpp.o src/libkiwix.so.13.1.0.p/version.cpp.o src/libkiwix.so.13.1.0.p/subprocess_unix.cpp.o -Wl,--as-needed -Wl,--no-undefined -shared -fPIC -Wl,-soname,libkiwix.so.13 -pthread -Wl,--start-group /usr/lib64/libicui18n.so /usr/lib64/libzim.so /usr/lib64/libpugixml.so /usr/lib64/libcurl.so /usr/lib64/libmicrohttpd.so /usr/lib64/libz.so /usr/lib64/libxapian.so -Wl,--end-group
/usr/lib64/gcc/x86_64-suse-linux/14/../../../../x86_64-suse-linux/bin/ld: src/libkiwix.so.13.1.0.p/library.cpp.o: in function `kiwix::(anonymous namespace)::iso639_3ToXapian(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)':
/home/aaron/vcs/libkiwix-13.1.0/build/../src/library.cpp:48:(.text+0xbd): undefined reference to `icu_77::Locale::Locale(char const*, char const*, char const*, char const*)'
/usr/lib64/gcc/x86_64-suse-linux/14/../../../../x86_64-suse-linux/bin/ld: /home/aaron/vcs/libkiwix-13.1.0/build/../src/library.cpp:48:(.text+0xf8): undefined reference to `icu_77::Locale::~Locale()'
/usr/lib64/gcc/x86_64-suse-linux/14/../../../../x86_64-suse-linux/bin/ld: /home/aaron/vcs/libkiwix-13.1.0/build/../src/library.cpp:48:(.text+0x11c): undefined reference to `icu_77::Locale::~Locale()'
/usr/lib64/gcc/x86_64-suse-linux/14/../../../../x86_64-suse-linux/bin/ld: src/libkiwix.so.13.1.0.p/tools_regexTools.cpp.o: in function `buildMatcher(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, icu_77::UnicodeString&)':
/home/aaron/vcs/libkiwix-13.1.0/build/../src/tools/regexTools.cpp:47:(.text+0x163): undefined reference to `icu_77::UnicodeString::UnicodeString(char const*)'
/usr/lib64/gcc/x86_64-suse-linux/14/../../../../x86_64-suse-linux/bin/ld: /home/aaron/vcs/libkiwix-13.1.0/build/../src/tools/regexTools.cpp:50:(.text+0x1ce): undefined reference to `icu_77::UnicodeString::~UnicodeString()'
/usr/lib64/gcc/x86_64-suse-linux/14/../../../../x86_64-suse-linux/bin/ld: /home/aaron/vcs/libkiwix-13.1.0/build/../src/tools/regexTools.cpp:50:(.text+0x1ea): undefined reference to `icu_77::UnicodeString::~UnicodeString()'
/usr/lib64/gcc/x86_64-suse-linux/14/../../../../x86_64-suse-linux/bin/ld: src/libkiwix.so.13.1.0.p/tools_regexTools.cpp.o: in function `matchRegex(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)':
/home/aaron/vcs/libkiwix-13.1.0/build/../src/tools/regexTools.cpp:58:(.text+0x253): undefined reference to `ucnv_setDefaultName_77'
/usr/lib64/gcc/x86_64-suse-linux/14/../../../../x86_64-suse-linux/bin/ld: /home/aaron/vcs/libkiwix-13.1.0/build/../src/tools/regexTools.cpp:59:(.text+0x271): undefined reference to `icu_77::UnicodeString::UnicodeString(char const*)'
/usr/lib64/gcc/x86_64-suse-linux/14/../../../../x86_64-suse-linux/bin/ld: /home/aaron/vcs/libkiwix-13.1.0/build/../src/tools/regexTools.cpp:62:(.text+0x2b9): undefined reference to `icu_77::UnicodeString::~UnicodeString()'
/usr/lib64/gcc/x86_64-suse-linux/14/../../../../x86_64-suse-linux/bin/ld: /home/aaron/vcs/libkiwix-13.1.0/build/../src/tools/regexTools.cpp:62:(.text+0x2dd): undefined reference to `icu_77::UnicodeString::~UnicodeString()'
/usr/lib64/gcc/x86_64-suse-linux/14/../../../../x86_64-suse-linux/bin/ld: src/libkiwix.so.13.1.0.p/tools_regexTools.cpp.o: in function `replaceRegex(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&, std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)':
/home/aaron/vcs/libkiwix-13.1.0/build/../src/tools/regexTools.cpp:68:(.text+0x325): undefined reference to `ucnv_setDefaultName_77'
/usr/lib64/gcc/x86_64-suse-linux/14/../../../../x86_64-suse-linux/bin/ld: /home/aaron/vcs/libkiwix-13.1.0/build/../src/tools/regexTools.cpp:69:(.text+0x346): undefined reference to `icu_77::UnicodeString::UnicodeString(char const*)'
/usr/lib64/gcc/x86_64-suse-linux/14/../../../../x86_64-suse-linux/bin/ld: /home/aaron/vcs/libkiwix-13.1.0/build/../src/tools/regexTools.cpp:70:(.text+0x36a): undefined reference to `icu_77::UnicodeString::UnicodeString(char const*)'
/usr/lib64/gcc/x86_64-suse-linux/14/../../../../x86_64-suse-linux/bin/ld: /home/aaron/vcs/libkiwix-13.1.0/build/../src/tools/regexTools.cpp:77:(.text+0x3f7): undefined reference to `icu_77::UnicodeString::~UnicodeString()'
/usr/lib64/gcc/x86_64-suse-linux/14/../../../../x86_64-suse-linux/bin/ld: /home/aaron/vcs/libkiwix-13.1.0/build/../src/tools/regexTools.cpp:77:(.text+0x415): undefined reference to `icu_77::UnicodeString::~UnicodeString()'
/usr/lib64/gcc/x86_64-suse-linux/14/../../../../x86_64-suse-linux/bin/ld: /home/aaron/vcs/libkiwix-13.1.0/build/../src/tools/regexTools.cpp:77:(.text+0x421): undefined reference to `icu_77::UnicodeString::~UnicodeString()'
/usr/lib64/gcc/x86_64-suse-linux/14/../../../../x86_64-suse-linux/bin/ld: /home/aaron/vcs/libkiwix-13.1.0/build/../src/tools/regexTools.cpp:77:(.text+0x444): undefined reference to `icu_77::UnicodeString::~UnicodeString()'
/usr/lib64/gcc/x86_64-suse-linux/14/../../../../x86_64-suse-linux/bin/ld: /home/aaron/vcs/libkiwix-13.1.0/build/../src/tools/regexTools.cpp:77:(.text+0x46c): undefined reference to `icu_77::UnicodeString::~UnicodeString()'
/usr/lib64/gcc/x86_64-suse-linux/14/../../../../x86_64-suse-linux/bin/ld: src/libkiwix.so.13.1.0.p/tools_regexTools.cpp.o:/home/aaron/vcs/libkiwix-13.1.0/build/../src/tools/regexTools.cpp:77: more undefined references to `icu_77::UnicodeString::~UnicodeString()' follow
/usr/lib64/gcc/x86_64-suse-linux/14/../../../../x86_64-suse-linux/bin/ld: src/libkiwix.so.13.1.0.p/tools_regexTools.cpp.o: in function `icu_77::ByteSink::ByteSink()':
/usr/include/unicode/bytestream.h:59:(.text._ZN6icu_778ByteSinkC2Ev[_ZN6icu_778ByteSinkC5Ev]+0xb): undefined reference to `vtable for icu_77::ByteSink'
/usr/lib64/gcc/x86_64-suse-linux/14/../../../../x86_64-suse-linux/bin/ld: src/libkiwix.so.13.1.0.p/tools_regexTools.cpp.o: in function `icu_77::StringByteSink<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >::~StringByteSink()':
/usr/include/unicode/bytestream.h:267:(.text._ZN6icu_7714StringByteSinkINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEED2Ev[_ZN6icu_7714StringByteSinkINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEED5Ev]+0x26): undefined reference to `icu_77::ByteSink::~ByteSink()'
/usr/lib64/gcc/x86_64-suse-linux/14/../../../../x86_64-suse-linux/bin/ld: src/libkiwix.so.13.1.0.p/tools_regexTools.cpp.o: in function `icu_77::StringByteSink<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >::~StringByteSink()':
/usr/include/unicode/bytestream.h:267:(.text._ZN6icu_7714StringByteSinkINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEED0Ev[_ZN6icu_7714StringByteSinkINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEED5Ev]+0x20): undefined reference to `icu_77::UMemory::operator delete(void*)'
/usr/lib64/gcc/x86_64-suse-linux/14/../../../../x86_64-suse-linux/bin/ld: src/libkiwix.so.13.1.0.p/tools_regexTools.cpp.o: in function `std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >& icu_77::UnicodeString::toUTF8String<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&) const':
/usr/include/unicode/unistr.h:1779:(.text._ZNK6icu_7713UnicodeString12toUTF8StringINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEEERT_S9_[_ZNK6icu_7713UnicodeString12toUTF8StringINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEEERT_S9_]+0x41): undefined reference to `icu_77::UnicodeString::toUTF8(icu_77::ByteSink&) const'
/usr/lib64/gcc/x86_64-suse-linux/14/../../../../x86_64-suse-linux/bin/ld: src/libkiwix.so.13.1.0.p/tools_regexTools.cpp.o: in function `icu_77::StringByteSink<std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > >::StringByteSink(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >*, int)':
/usr/include/unicode/bytestream.h:287:(.text._ZN6icu_7714StringByteSinkINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEEC2EPS6_i[_ZN6icu_7714StringByteSinkINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEEC5EPS6_i]+0xb3): undefined reference to `icu_77::ByteSink::~ByteSink()'
/usr/lib64/gcc/x86_64-suse-linux/14/../../../../x86_64-suse-linux/bin/ld: src/libkiwix.so.13.1.0.p/tools_regexTools.cpp.o:(.data.rel.ro._ZTVN6icu_7714StringByteSinkINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEEE[_ZTVN6icu_7714StringByteSinkINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEEE]+0x28): undefined reference to `icu_77::ByteSink::GetAppendBuffer(int, int, char*, int, int*)'
/usr/lib64/gcc/x86_64-suse-linux/14/../../../../x86_64-suse-linux/bin/ld: src/libkiwix.so.13.1.0.p/tools_regexTools.cpp.o:(.data.rel.ro._ZTVN6icu_7714StringByteSinkINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEEE[_ZTVN6icu_7714StringByteSinkINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEEE]+0x30): undefined reference to `icu_77::ByteSink::Flush()'
/usr/lib64/gcc/x86_64-suse-linux/14/../../../../x86_64-suse-linux/bin/ld: src/libkiwix.so.13.1.0.p/tools_regexTools.cpp.o:(.data.rel.ro._ZTIN6icu_7714StringByteSinkINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEEE[_ZTIN6icu_7714StringByteSinkINSt7__cxx1112basic_stringIcSt11char_traitsIcESaIcEEEEE]+0x10): undefined reference to `typeinfo for icu_77::ByteSink'
/usr/lib64/gcc/x86_64-suse-linux/14/../../../../x86_64-suse-linux/bin/ld: src/libkiwix.so.13.1.0.p/tools_stringTools.cpp.o: in function `kiwix::ICULanguageInfo::ICULanguageInfo(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)':
/home/aaron/vcs/libkiwix-13.1.0/build/../src/tools/stringTools.cpp:53:(.text+0x40): undefined reference to `icu_77::Locale::Locale(char const*, char const*, char const*, char const*)'
/usr/lib64/gcc/x86_64-suse-linux/14/../../../../x86_64-suse-linux/bin/ld: src/libkiwix.so.13.1.0.p/tools_stringTools.cpp.o: in function `kiwix::ICULanguageInfo::iso3Code[abi:cxx11]() const':
/home/aaron/vcs/libkiwix-13.1.0/build/../src/tools/stringTools.cpp:58:(.text+0x6f): undefined reference to `icu_77::Locale::getISO3Language() const'
/usr/lib64/gcc/x86_64-suse-linux/14/../../../../x86_64-suse-linux/bin/ld: src/libkiwix.so.13.1.0.p/tools_stringTools.cpp.o: in function `kiwix::ICULanguageInfo::selfName[abi:cxx11]() const':
/home/aaron/vcs/libkiwix-13.1.0/build/../src/tools/stringTools.cpp:64:(.text+0xee): undefined reference to `icu_77::Locale::getDisplayLanguage(icu_77::Locale const&, icu_77::UnicodeString&) const'
/usr/lib64/gcc/x86_64-suse-linux/14/../../../../x86_64-suse-linux/bin/ld: /home/aaron/vcs/libkiwix-13.1.0/build/../src/tools/stringTools.cpp:68:(.text+0x119): undefined reference to `icu_77::UnicodeString::~UnicodeString()'
/usr/lib64/gcc/x86_64-suse-linux/14/../../../../x86_64-suse-linux/bin/ld: /home/aaron/vcs/libkiwix-13.1.0/build/../src/tools/stringTools.cpp:68:(.text+0x13b): undefined reference to `icu_77::UnicodeString::~UnicodeString()'
/usr/lib64/gcc/x86_64-suse-linux/14/../../../../x86_64-suse-linux/bin/ld: src/libkiwix.so.13.1.0.p/tools_stringTools.cpp.o: in function `kiwix::removeAccents(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)':
/home/aaron/vcs/libkiwix-13.1.0/build/../src/tools/stringTools.cpp:73:(.text+0x17e): undefined reference to `ucnv_setDefaultName_77'
/usr/lib64/gcc/x86_64-suse-linux/14/../../../../x86_64-suse-linux/bin/ld: /home/aaron/vcs/libkiwix-13.1.0/build/../src/tools/stringTools.cpp:75:(.text+0x19b): undefined reference to `icu_77::UnicodeString::UnicodeString(char const*)'
/usr/lib64/gcc/x86_64-suse-linux/14/../../../../x86_64-suse-linux/bin/ld: /home/aaron/vcs/libkiwix-13.1.0/build/../src/tools/stringTools.cpp:75:(.text+0x1c0): undefined reference to `icu_77::UnicodeString::~UnicodeString()'
/usr/lib64/gcc/x86_64-suse-linux/14/../../../../x86_64-suse-linux/bin/ld: /home/aaron/vcs/libkiwix-13.1.0/build/../src/tools/stringTools.cpp:77:(.text+0x1e4): undefined reference to `icu_77::UnicodeString::UnicodeString(char const*)'
/usr/lib64/gcc/x86_64-suse-linux/14/../../../../x86_64-suse-linux/bin/ld: /home/aaron/vcs/libkiwix-13.1.0/build/../src/tools/stringTools.cpp:83:(.text+0x254): undefined reference to `icu_77::UnicodeString::~UnicodeString()'
/usr/lib64/gcc/x86_64-suse-linux/14/../../../../x86_64-suse-linux/bin/ld: /home/aaron/vcs/libkiwix-13.1.0/build/../src/tools/stringTools.cpp:75:(.text+0x265): undefined reference to `icu_77::UnicodeString::~UnicodeString()'
/usr/lib64/gcc/x86_64-suse-linux/14/../../../../x86_64-suse-linux/bin/ld: /home/aaron/vcs/libkiwix-13.1.0/build/../src/tools/stringTools.cpp:83:(.text+0x296): undefined reference to `icu_77::UnicodeString::~UnicodeString()'
/usr/lib64/gcc/x86_64-suse-linux/14/../../../../x86_64-suse-linux/bin/ld: src/libkiwix.so.13.1.0.p/tools_stringTools.cpp.o: in function `kiwix::printStringInHexadecimal(icu_77::UnicodeString)':
/home/aaron/vcs/libkiwix-13.1.0/build/../src/tools/stringTools.cpp:120:(.text+0x655): undefined reference to `icu_77::UnicodeString::getTerminatedBuffer()'
/usr/lib64/gcc/x86_64-suse-linux/14/../../../../x86_64-suse-linux/bin/ld: src/libkiwix.so.13.1.0.p/tools_stringTools.cpp.o: in function `kiwix::ucFirst(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)':
/home/aaron/vcs/libkiwix-13.1.0/build/../src/tools/stringTools.cpp:339:(.text+0x1480): undefined reference to `icu_77::UnicodeString::UnicodeString(char const*)'
/usr/lib64/gcc/x86_64-suse-linux/14/../../../../x86_64-suse-linux/bin/ld: /home/aaron/vcs/libkiwix-13.1.0/build/../src/tools/stringTools.cpp:340:(.text+0x149d): undefined reference to `icu_77::UnicodeString::UnicodeString(icu_77::UnicodeString const&, int, int)'
/usr/lib64/gcc/x86_64-suse-linux/14/../../../../x86_64-suse-linux/bin/ld: /home/aaron/vcs/libkiwix-13.1.0/build/../src/tools/stringTools.cpp:340:(.text+0x14a9): undefined reference to `icu_77::UnicodeString::toUpper()'
/usr/lib64/gcc/x86_64-suse-linux/14/../../../../x86_64-suse-linux/bin/ld: /home/aaron/vcs/libkiwix-13.1.0/build/../src/tools/stringTools.cpp:340:(.text+0x14be): undefined reference to `icu_77::UnicodeString::UnicodeString(icu_77::UnicodeString const&)'
/usr/lib64/gcc/x86_64-suse-linux/14/../../../../x86_64-suse-linux/bin/ld: /home/aaron/vcs/libkiwix-13.1.0/build/../src/tools/stringTools.cpp:340:(.text+0x14ca): undefined reference to `icu_77::UnicodeString::~UnicodeString()'
/usr/lib64/gcc/x86_64-suse-linux/14/../../../../x86_64-suse-linux/bin/ld: /home/aaron/vcs/libkiwix-13.1.0/build/../src/tools/stringTools.cpp:345:(.text+0x152e): undefined reference to `icu_77::UnicodeString::~UnicodeString()'
/usr/lib64/gcc/x86_64-suse-linux/14/../../../../x86_64-suse-linux/bin/ld: /home/aaron/vcs/libkiwix-13.1.0/build/../src/tools/stringTools.cpp:345:(.text+0x153d): undefined reference to `icu_77::UnicodeString::~UnicodeString()'
/usr/lib64/gcc/x86_64-suse-linux/14/../../../../x86_64-suse-linux/bin/ld: /home/aaron/vcs/libkiwix-13.1.0/build/../src/tools/stringTools.cpp:340:(.text+0x1578): undefined reference to `icu_77::UnicodeString::~UnicodeString()'
/usr/lib64/gcc/x86_64-suse-linux/14/../../../../x86_64-suse-linux/bin/ld: /home/aaron/vcs/libkiwix-13.1.0/build/../src/tools/stringTools.cpp:345:(.text+0x158c): undefined reference to `icu_77::UnicodeString::~UnicodeString()'
/usr/lib64/gcc/x86_64-suse-linux/14/../../../../x86_64-suse-linux/bin/ld: src/libkiwix.so.13.1.0.p/tools_stringTools.cpp.o:/home/aaron/vcs/libkiwix-13.1.0/build/../src/tools/stringTools.cpp:345: more undefined references to `icu_77::UnicodeString::~UnicodeString()' follow
/usr/lib64/gcc/x86_64-suse-linux/14/../../../../x86_64-suse-linux/bin/ld: src/libkiwix.so.13.1.0.p/tools_stringTools.cpp.o: in function `kiwix::ucAll(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)':
/home/aaron/vcs/libkiwix-13.1.0/build/../src/tools/stringTools.cpp:355:(.text+0x165e): undefined reference to `icu_77::UnicodeString::UnicodeString(char const*)'
/usr/lib64/gcc/x86_64-suse-linux/14/../../../../x86_64-suse-linux/bin/ld: /home/aaron/vcs/libkiwix-13.1.0/build/../src/tools/stringTools.cpp:356:(.text+0x166a): undefined reference to `icu_77::UnicodeString::toUpper()'
/usr/lib64/gcc/x86_64-suse-linux/14/../../../../x86_64-suse-linux/bin/ld: /home/aaron/vcs/libkiwix-13.1.0/build/../src/tools/stringTools.cpp:359:(.text+0x169e): undefined reference to `icu_77::UnicodeString::~UnicodeString()'
/usr/lib64/gcc/x86_64-suse-linux/14/../../../../x86_64-suse-linux/bin/ld: /home/aaron/vcs/libkiwix-13.1.0/build/../src/tools/stringTools.cpp:359:(.text+0x16d6): undefined reference to `icu_77::UnicodeString::~UnicodeString()'
/usr/lib64/gcc/x86_64-suse-linux/14/../../../../x86_64-suse-linux/bin/ld: src/libkiwix.so.13.1.0.p/tools_stringTools.cpp.o: in function `kiwix::lcFirst(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)':
/home/aaron/vcs/libkiwix-13.1.0/build/../src/tools/stringTools.cpp:369:(.text+0x1797): undefined reference to `icu_77::UnicodeString::UnicodeString(char const*)'
/usr/lib64/gcc/x86_64-suse-linux/14/../../../../x86_64-suse-linux/bin/ld: /home/aaron/vcs/libkiwix-13.1.0/build/../src/tools/stringTools.cpp:370:(.text+0x17b4): undefined reference to `icu_77::UnicodeString::UnicodeString(icu_77::UnicodeString const&, int, int)'
/usr/lib64/gcc/x86_64-suse-linux/14/../../../../x86_64-suse-linux/bin/ld: /home/aaron/vcs/libkiwix-13.1.0/build/../src/tools/stringTools.cpp:370:(.text+0x17c0): undefined reference to `icu_77::UnicodeString::toLower()'
/usr/lib64/gcc/x86_64-suse-linux/14/../../../../x86_64-suse-linux/bin/ld: /home/aaron/vcs/libkiwix-13.1.0/build/../src/tools/stringTools.cpp:370:(.text+0x17d5): undefined reference to `icu_77::UnicodeString::UnicodeString(icu_77::UnicodeString const&)'
/usr/lib64/gcc/x86_64-suse-linux/14/../../../../x86_64-suse-linux/bin/ld: /home/aaron/vcs/libkiwix-13.1.0/build/../src/tools/stringTools.cpp:370:(.text+0x17e1): undefined reference to `icu_77::UnicodeString::~UnicodeString()'
/usr/lib64/gcc/x86_64-suse-linux/14/../../../../x86_64-suse-linux/bin/ld: /home/aaron/vcs/libkiwix-13.1.0/build/../src/tools/stringTools.cpp:375:(.text+0x1845): undefined reference to `icu_77::UnicodeString::~UnicodeString()'
/usr/lib64/gcc/x86_64-suse-linux/14/../../../../x86_64-suse-linux/bin/ld: /home/aaron/vcs/libkiwix-13.1.0/build/../src/tools/stringTools.cpp:375:(.text+0x1854): undefined reference to `icu_77::UnicodeString::~UnicodeString()'
/usr/lib64/gcc/x86_64-suse-linux/14/../../../../x86_64-suse-linux/bin/ld: /home/aaron/vcs/libkiwix-13.1.0/build/../src/tools/stringTools.cpp:370:(.text+0x188f): undefined reference to `icu_77::UnicodeString::~UnicodeString()'
/usr/lib64/gcc/x86_64-suse-linux/14/../../../../x86_64-suse-linux/bin/ld: /home/aaron/vcs/libkiwix-13.1.0/build/../src/tools/stringTools.cpp:375:(.text+0x18a3): undefined reference to `icu_77::UnicodeString::~UnicodeString()'
/usr/lib64/gcc/x86_64-suse-linux/14/../../../../x86_64-suse-linux/bin/ld: src/libkiwix.so.13.1.0.p/tools_stringTools.cpp.o:/home/aaron/vcs/libkiwix-13.1.0/build/../src/tools/stringTools.cpp:375: more undefined references to `icu_77::UnicodeString::~UnicodeString()' follow
/usr/lib64/gcc/x86_64-suse-linux/14/../../../../x86_64-suse-linux/bin/ld: src/libkiwix.so.13.1.0.p/tools_stringTools.cpp.o: in function `kiwix::lcAll(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)':
/home/aaron/vcs/libkiwix-13.1.0/build/../src/tools/stringTools.cpp:385:(.text+0x1975): undefined reference to `icu_77::UnicodeString::UnicodeString(char const*)'
/usr/lib64/gcc/x86_64-suse-linux/14/../../../../x86_64-suse-linux/bin/ld: /home/aaron/vcs/libkiwix-13.1.0/build/../src/tools/stringTools.cpp:386:(.text+0x1981): undefined reference to `icu_77::UnicodeString::toLower()'
/usr/lib64/gcc/x86_64-suse-linux/14/../../../../x86_64-suse-linux/bin/ld: /home/aaron/vcs/libkiwix-13.1.0/build/../src/tools/stringTools.cpp:389:(.text+0x19b5): undefined reference to `icu_77::UnicodeString::~UnicodeString()'
/usr/lib64/gcc/x86_64-suse-linux/14/../../../../x86_64-suse-linux/bin/ld: /home/aaron/vcs/libkiwix-13.1.0/build/../src/tools/stringTools.cpp:389:(.text+0x19ed): undefined reference to `icu_77::UnicodeString::~UnicodeString()'
/usr/lib64/gcc/x86_64-suse-linux/14/../../../../x86_64-suse-linux/bin/ld: src/libkiwix.so.13.1.0.p/tools_stringTools.cpp.o: in function `kiwix::toTitle(std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> > const&)':
/home/aaron/vcs/libkiwix-13.1.0/build/../src/tools/stringTools.cpp:399:(.text+0x1aa8): undefined reference to `icu_77::UnicodeString::UnicodeString(char const*)'
/usr/lib64/gcc/x86_64-suse-linux/14/../../../../x86_64-suse-linux/bin/ld: /home/aaron/vcs/libkiwix-13.1.0/build/../src/tools/stringTools.cpp:400:(.text+0x1ab9): undefined reference to `icu_77::UnicodeString::toTitle(icu_77::BreakIterator*)'
/usr/lib64/gcc/x86_64-suse-linux/14/../../../../x86_64-suse-linux/bin/ld: /home/aaron/vcs/libkiwix-13.1.0/build/../src/tools/stringTools.cpp:400:(.text+0x1acb): undefined reference to `icu_77::UnicodeString::operator=(icu_77::UnicodeString const&)'
/usr/lib64/gcc/x86_64-suse-linux/14/../../../../x86_64-suse-linux/bin/ld: /home/aaron/vcs/libkiwix-13.1.0/build/../src/tools/stringTools.cpp:404:(.text+0x1b00): undefined reference to `icu_77::UnicodeString::~UnicodeString()'
/usr/lib64/gcc/x86_64-suse-linux/14/../../../../x86_64-suse-linux/bin/ld: /home/aaron/vcs/libkiwix-13.1.0/build/../src/tools/stringTools.cpp:404:(.text+0x1b38): undefined reference to `icu_77::UnicodeString::~UnicodeString()'
/usr/lib64/gcc/x86_64-suse-linux/14/../../../../x86_64-suse-linux/bin/ld: src/libkiwix.so.13.1.0.p/tools_stringTools.cpp.o: in function `icu_77::UObject::UObject()':
/usr/include/unicode/uobject.h:223:(.text._ZN6icu_777UObjectC2Ev[_ZN6icu_777UObjectC5Ev]+0xb): undefined reference to `vtable for icu_77::UObject'
/usr/lib64/gcc/x86_64-suse-linux/14/../../../../x86_64-suse-linux/bin/ld: src/libkiwix.so.13.1.0.p/tools_stringTools.cpp.o: in function `icu_77::Replaceable::Replaceable()':
/usr/include/unicode/rep.h:243:(.text._ZN6icu_7711ReplaceableC2Ev[_ZN6icu_7711ReplaceableC5Ev]+0x1b): undefined reference to `vtable for icu_77::Replaceable'
/usr/lib64/gcc/x86_64-suse-linux/14/../../../../x86_64-suse-linux/bin/ld: src/libkiwix.so.13.1.0.p/tools_stringTools.cpp.o: in function `icu_77::UnicodeString::UnicodeString()':
/usr/include/unicode/unistr.h:4182:(.text._ZN6icu_7713UnicodeStringC2Ev[_ZN6icu_7713UnicodeStringC5Ev]+0x1b): undefined reference to `vtable for icu_77::UnicodeString'
/usr/lib64/gcc/x86_64-suse-linux/14/../../../../x86_64-suse-linux/bin/ld: src/libkiwix.so.13.1.0.p/tools_stringTools.cpp.o: in function `icu_77::UnicodeString::replace(int, int, icu_77::UnicodeString const&)':
/usr/include/unicode/unistr.h:4733:(.text._ZN6icu_7713UnicodeString7replaceEiiRKS0_[_ZN6icu_7713UnicodeString7replaceEiiRKS0_]+0x3f): undefined reference to `icu_77::UnicodeString::doReplace(int, int, icu_77::UnicodeString const&, int, int)'
/usr/lib64/gcc/x86_64-suse-linux/14/../../../../x86_64-suse-linux/bin/ld: src/libkiwix.so.13.1.0.p/tools_languageTools.cpp.o: in function `kiwix::(anonymous namespace)::fillLanguagesMap()':
/home/aaron/vcs/libkiwix-13.1.0/build/../src/tools/languageTools.cpp:57:(.text+0xd): undefined reference to `icu_77::Locale::getISOLanguages()'
/usr/lib64/gcc/x86_64-suse-linux/14/../../../../x86_64-suse-linux/bin/ld: src/libkiwix.so.13.1.0.p/tools_languageTools.cpp.o: in function `kiwix::ICULanguageInfo::~ICULanguageInfo()':
/home/aaron/vcs/libkiwix-13.1.0/build/../src/tools/stringTools.h:44:(.text._ZN5kiwix15ICULanguageInfoD2Ev[_ZN5kiwix15ICULanguageInfoD5Ev]+0x14): undefined reference to `icu_77::Locale::~Locale()'
collect2: error: ld returned 1 exit status
[93/121] Compiling C++ object test/server.p/server.cpp.o
ninja: build stopped: subcommand failed.
```

as stringTools tries to use Locale, which isn't in i18n.

I was able to work around this by renaming things on `meson.build`. However, I don't know meson well enough to say if there is a more proper way to do this.

```patch
--- ./meson.build.orig  2025-08-30 14:17:22.211369285 -0400
+++ ./meson.build       2025-08-30 14:18:45.727383770 -0400
@@ -20,7 +20,8 @@
 else
   thread_dep = dependency('', required:false)
 endif
-libicu_dep = dependency('icu-i18n', static:static_deps)
+libicuuc_dep = dependency('icu-uc', static:static_deps)
+libicui18n_dep = dependency('icu-i18n', static:static_deps)
 pugixml_dep = dependency('pugixml', static:static_deps)
 libcurl_dep = dependency('libcurl', static:static_deps)
 microhttpd_dep = dependency('libmicrohttpd', static:static_deps)
@@ -51,7 +52,7 @@
   add_project_arguments('-DNOMINMAX', language: 'cpp')
 endif

-all_deps = [thread_dep, libicu_dep, libzim_dep, pugixml_dep, libcurl_dep, microhttpd_dep, zlib_dep, xapian_dep]
+all_deps = [thread_dep, libicuuc_dep, libicui18n_dep, libzim_dep, pugixml_dep, libcurl_dep, microhttpd_dep, zlib_dep, xapian_dep]

 inc = include_directories('include', extra_include)
```

--- a/meson.build
+++ b/meson.build
@@ -48,7 +48,14 @@ if (host_machine.system() == 'linux' and compiler.get_id() == 'gcc') or \
 else
   thread_dep = dependency('', required:false)
 endif
+
 libicu_dep = dependency('icu-i18n', static:static_deps)
+if libicu_dep.version().version_compare('>= 76')
+  libicu_deps = [libicu_dep, dependency('icu-uc', static:static_deps)]
+else
+  libicu_deps = [libicu_dep]
+endif
+
 pugixml_dep = dependency('pugixml', static:static_deps)
 libcurl_dep = dependency('libcurl', static:static_deps)
 microhttpd_dep = dependency('libmicrohttpd', static:static_deps)
@@ -86,7 +93,11 @@ if build_machine.system() == 'windows'
 endif
 
 
-all_deps = [thread_dep, libicu_dep, libzim_dep, pugixml_dep, libcurl_dep, microhttpd_dep, zlib_dep, xapian_dep]
+# Dependencies as string
+all_deps = [thread_dep, libzim_dep, pugixml_dep, libcurl_dep, microhttpd_dep, zlib_dep, xapian_dep]
+
+# Dependencies as array
+all_deps += libicu_deps
 
 inc = include_directories('include', extra_include)
 

